src/components/structures/MatrixChat.tsx

```diff
@@ -1994,18 +1995,34 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
                  * as using something like redux to avoid having a billion bits of state kicking around.
                  */
                 const LoggedInView = sdk.getComponent('structures.LoggedInView');
-                view = (
-                    <LoggedInView
-                        {...this.props}
-                        {...this.state}
-                        ref={this.loggedInView}
-                        matrixClient={MatrixClientPeg.get()}
-                        onRoomCreated={this.onRoomCreated}
-                        onCloseAllSettings={this.onCloseAllSettings}
-                        onRegistered={this.onRegistered}
-                        currentRoomId={this.state.currentRoomId}
-                    />
-                );
+
+                const roleSelector = SdkConfig.get()["role_selector"];//"role-selection"
+                const roleSelectorText = roleSelector?.text;// SdkConfig.get()["role_selector_name"];//"role-selection"
+                const roleSelectorUrl = roleSelector?.url;//SdkConfig.get()["role_selector_url"];//'/role-selection/#'
+
+                let ref = document.referrer;
+                console.log(ref!= "");
+                console.log(ref?.toLowerCase()?.includes(roleSelectorText));
+                
+                if(ref !="" && ref?.toLowerCase()?.includes(roleSelectorText))
```

Comment:
Should this if be the following code to avoid not affecting the standard element behaviour:
```
if((!roleSelectorText) || (ref !="" && ref?.toLowerCase()?.includes(roleSelectorText)))
``` 
Also I don't understand why the ref !="" && check is required ... if ref == "" then ref?.toLowerCase()?.includes(roleSelectorText) would return false anyway

Reply:
Done

Reply:
Hi Samridhi, while you removed the "ref !="" && ", you didn't add (!roleSelectorText) || to avoid not affecting the standard element behaviour.  Thanks Jasen